certbot/certbot/util.py

```diff
@@ -48,6 +50,200 @@ class CSR(NamedTuple):
+        version have not yet been compared. If these are all equal to zero, the two versions are
+        equal. Otherwise, the longer version is greater.
+
+        If the two versions are incomparible, an exception is raised. Otherwise, the returned
+        integer indicates the result of the comparison. If self == other, 0 is returned.
+        If self > other, 1 is returned. If self < other -1 is returned.
+
+        Examples:
+        Equality:
+        - LooseVersion('1.0').try_risky_comparison(LooseVersion('1.0')) -> 0
+        - LooseVersion('2.0.0a').try_risky_comparison(LooseVersion('2.0.0a')) -> 0
+        Inequality:
+        - LooseVersion('2.0.0').try_risky_comparison(LooseVersion('1.0')) -> 1
+        - LooseVersion('1.0.1').try_risky_comparison(LooseVersion('2.0a')) -> -1
+        Incomparability:
+        - LooseVersion('1a').try_risky_comparison(LooseVersion('1.0')) -> ValueError
+        """
+        if not isinstance(other, type(self)):
+            raise TypeError(f'Comparison not supported between LooseVersion and {type(other)}')
+
+        comparison = self.__compare(other)
+
+        if comparison == self.Comparison.EQUAL:
+            return 0
+        elif comparison == self.Comparison.LESS:
+            return -1
+        elif comparison == self.Comparison.GREATER:
+            return 1
+        else:
+            raise ValueError("Cannot meaningfully compare LooseVersion {} with LooseVersion {} "
+                             "due to comparison of version components with different types."
+                             .format(self.version_components, other.version_components))
+
+    # Provide helper methods for comparison without having to use the returned integer value.
+    def try_risky_equal(self, other: Any) -> bool:
+        """Tests if self == other, using `try_risky_comparison`.
+        """
+        return self.try_risky_comparison(other) == 0
+
+    def try_risky_not_equal(self, other: Any) -> bool:
+        """Tests if self != other, using `try_risky_comparison`.
+        """
+        return self.try_risky_comparison(other) != 0
+
+    def try_risky_less(self, other: Any) -> bool:
+        """Tests if self < other, using `try_risky_comparison`.
+        """
+        return self.try_risky_comparison(other) == -1
+
+    def try_risky_greater(self, other: Any) -> bool:
+        """Tests if self > other, using `try_risky_comparison`.
+        """
+        return self.try_risky_comparison(other) == 1
+
+    def try_risky_less_equal(self, other: Any) -> bool:
+        """Tests if self <= other, using `try_risky_comparison`.
+        """
+        return self.try_risky_comparison(other) in (0, -1)
+
+    def try_risky_greater_equal(self, other: Any) -> bool:
+        """Tests if self >= other, using `try_risky_comparison`.
+        """
+        return self.try_risky_comparison(other) in (0, 1)
+
+    # Prevent the use of regular comparison operators
```

Comment:
I also don't think we should provide these. With them, `mypy` doesn't complain about trying to compare two `LooseVersion` objects and without them it does.